src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/PackageDeclarationCheck.java

```diff
@@ -140,11 +137,12 @@ public void beginTree(DetailAST ast) {
     @Override
     public void finishTree(DetailAST ast) {
         if (!defined) {
-            int lineNumber = DEFAULT_LINE_NUMBER;
-            if (ast != null) {
-                lineNumber = ast.getLineNo();
+            if (ast == null) {
+                log(1, 0, MSG_KEY_MISSING);
```

Comment:
it looks like this can lead to an `ArrayIndexOutOfBoundsException` on empty files - is that a concern?

* https://dxtkastb.github.io/checkstyle/reports/packed/diff/hibernate-orm/index.html#A19
* https://dxtkastb.github.io/checkstyle/reports/packed/diff/openjdk9/index.html#A38033

Reply:
Yes, one of the tests for this check is made on the empty java file(with at least one comment). Due to this, Ast goes null and so we are using `log(1, 0, MSG_KEY_MISSING);`. This points violation on line:1 coumn:1 of the file.

The issue here is empty file can look like:
1.Single line comment in the file
2.Multiline comment
3.Completely empty file..No comment nothing.

The first two cases are covered but if it is completely empty its a problem. I will add a test for this too. Thanks for pointing this out.

Reply:
Maybe in that third case we should fall back on the old way of logging with only a line number rather than throw an exception?

Reply:
I think I made a mistake by passing '0' as column number, as I did not pay heed to this empty file scenario. And yes I should probably switch back to old logging way otherwise the exception has to be caught and the method for processing files in Checker.java will need to be changed drastically which is not a wise move.

Reply:
I will do the changes suggested and provide the new diff report shortly. Once again, thanks for the guidance.

Reply:
Sounds good, thanks @DXTkastb!